packages/astro/src/core/build/generate.ts

```diff
@@ -231,7 +232,7 @@ async function getPathsForRoute(
 			route: pageData.route,
 			isValidate: false,
 			logging: opts.logging,
-			ssr: opts.settings.config.output === 'server',
+			ssr: opts.settings.config.output === 'server' || isHybridOutput(opts.settings.config),
```

Comment:
I could refactor these further with an `isSSR` utility but in my option it will make the code more difficult to understand, without stepping into the function.
Just my option, I'm open to changing it!